test/addons/dlopen-hello-world/test.js

```diff
@@ -0,0 +1,8 @@
+'use strict';
+const common = require('../../common');
+const assert = require('assert');
+const os = require('os');
+process.dlopen(module, require.resolve(`./build/${common.buildType}/binding`),
+               os.constants.dlopen.RTLD_NOW | os.constants.dlopen.RTLD_LOCAL)
```

Comment:
I guess this works by accident on Windows?  `undefined | undefined` is `NaN | NaN` is `0 | 0` is `0`.

In that respect this test doesn't really test anything, it will work whether the dlopen flags are respected or not.

Alternative: two add-ons, first one loaded with RTLD_GLOBAL, second one linking to a symbol defined in the first one (and skip the test on Windows.)